tide-whisperer.go

```diff
@@ -167,6 +168,44 @@ func main() {
 		log.Printf("%s mongo processing finished after %.5f seconds and returned %d records", DATA_API_PREFIX, time.Now().Sub(startTime).Seconds(), writeCount)
 	}
 
+	authorize := func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			start := time.Now()
+
+			var token string
+			if authorization := r.Header.Get("Authorization"); authorization != "" {
+				if parts := strings.Split(authorization, " "); len(parts) == 2 && strings.EqualFold(parts[0], "Bearer") {
+					log.Println("Validating with access_token")
+					token = parts[1]
+				}
+			}
+			if token == "" {
+				token = r.Header.Get("X-Tidepool-Session-Token")
+				if token == "" {
+					jsonError(w, error_no_view_permisson, start)
+					return
+				}
+			}
+
+			tokenData := shorelineClient.CheckToken(token)
+			if tokenData != nil {
+				queryParams, err := store.GetParams(r.URL.Query(), &config.SchemaVersion)
+				if err == nil {
+					if userCanViewData(tokenData.UserID, queryParams.UserId) ||
```

Comment:
I would flip the order of the conditional: if tokenData.isServer || userCanViewData(tokenData.UserID, queryParams.UserId)